### PR TITLE
[stable/gangway] add logic for applying env vars and pod annotations

### PR DESCRIPTION
**Pull request by cten**
…tations

Signed-off-by: cten <cxtenberge5@gmail.com>

#### What this PR does / why we need it:
I needed to pass pod annotations and environment variables to the pod. This allows for loading secrets from env. 

#### Which issue this PR fixes
I don't believe there is a ticket at this time

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

@rk295

### CODE REVIEW

**Comment by cten** [comment]
Any maintainers have time to help me merge this? Thanks

**Comment by cten** [comment]
Ping...

**Comment by rk295** [comment]
sorry for lag, things bit crazy atm!

**Comment by cten** [comment]
Thanks! Completely understand.

File: stable/gangway/templates/key.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.env.GANGWAY_SESSION_SECURITY_KEY -}}
```
Comment (rustycl0ck) [inline stable/gangway/templates/key.yaml:1]: This breaks the chart with error: `Error: Failed to render chart: exit status 1: Error: template: gangway/templates/key.yaml:1:18: executing "gangway/templates/key.yaml" at <.Values.env.GANGWAY_SESSION_SECURITY_KEY>: can't evaluate field GANGWAY_SESSION_SECURITY_KEY in type interface {}`

**Review by rustycl0ck — COMMENTED** [review]
The default installation of this chart is now broken.
EDIT: This is all good, I was trying to install with `env: []` instead of `env: {}`

File: stable/gangway/templates/key.yaml (hunk 1 above)
Comment (rustycl0ck) [reply stable/gangway/templates/key.yaml:1]: Sorry, I had passed `env: []` by mistake instead of the default `env: {}`. This works fine.

**Pull request merged**